lib/tools/adb-commands.js

```diff
@@ -277,8 +276,7 @@ methods.isWifiOn = async function () {
  * on: true (to turn on) or false (to turn off)
  */
 methods.setWifiState = async function (on) {
-  await this.shell(['am', 'start', '-n', 'io.appium.settings/.Settings', '-e',
-                    'wifi', on ? 'on' : 'off']);
+  await this.shell(['settings', 'put', 'global', 'wifi_on', on ? 1 : 0]);
```

Comment:
Yes, commands: "adb shell settings put global mobile_data 1" like this works fine for me.